File: Frequency Game.java
class Solution {
    public static int LargButMinFreq(int arr[], int n) {
        HashMap<Integer,Integer>map = new HashMap();
        for(int i:arr) map.put(i,map.getOrDefault(i,0)+1);
        int freq = Integer.MAX_VALUE,max = Integer.MIN_VALUE;
        for(Map.Entry<Integer,Integer>mp:map.entrySet()){
            int key = mp.getKey(),value = mp.getValue();
            if(freq>=value && key>=max){
                freq = value;
                max = key;
            }
        }
        return max;
    }
}

File: Frequency Game.cpp
class Solution{
public:
    int LargButMinFreq(int arr[], int n) {
        map<int,int>mp;
        for(int i=0;i<n;i++){
            mp[arr[i]]++;
        }
        int mini = INT_MAX;
        int minnum = -1;
        for(auto it : mp){
            if(it.second <= mini){
                mini = it.second;
                minnum = it.first;
            }
        }
        return  minnum;
    }
};
